src/DI/ContainerBuilder.php

```diff
@@ -355,12 +355,17 @@ private function resolveImplement(ServiceDefinition $def, $name)
 		$def->setImplementType($methodName = $rc->hasMethod('create') ? 'create' : 'get');
 
 		if (!$def->getClass() && !$def->getEntity()) {
-			$returnType = PhpReflection::parseAnnotation($method, 'return');
-			if (!$returnType) {
-				throw new ServiceCreationException("Method $interface::$methodName() used in service '$name' has no @return annotation.");
+			if (PHP_VERSION_ID >= 70000 && $method->hasReturnType()) {
+				$returnType = (string) $method->getReturnType();
+			} else {
```

Comment:
This should really have a `@return annotation fallback`, simple `else` will cause problems. This would be a BC Break - an unnecesary one.